### DISCUSSION TITLE
HeroMe Gen5 and direct drive settings

### DISCUSSION THREAD

**Discussion by breaktech** [Jyers/Marlin#908] [Q&A] [answered]
Hi, I wanted to test the herome fun duct on ender 3 v2 and while I was there I modified the gantry plate adapter to host extruder and stepper for direct drive.
Now I have a couple of problems:
1) Herome mounts the hotend 9mm further from the plate. I tried to use M206 Y-9 and M500 via octoprint, I get positive feedback from the printer but when I home and move to Y0 the nozzle is still out of the bed, it seems that the offset is not read...
2) I mounted the stepper on the back so I'm loosing approx 10mm on left and right side of X, I'm using a small 10mm extension piece to activate the limit switch and avoid slamming the motor.
When I level the bed though the motor still slams on the right...
Is it possible to reduce the distance of the mesh levelling matrix to avoid this?
Is it possible to tell the machine that max x is 202mm so to avoid collisions of the motor?

thanks!

**Comment by breaktech** [comment]
now, if I set positions via gcode in octoprint, say G0 X0 Y0 Z0 it works as expected and the offset is taken into account...
but if I use the gui on the printer Y is reported to be as "9" and if I move to Y0 via printer's GUI nozzle is again off the bed

**Comment by breaktech** [comment, marked as answer]
I managed to solve the problems above by modifying source code and recompiling marlin, while I was there I enabled linear advance to test it but something strange happens: when I set K=0 everything work as usual but changing to something else than 0 make the extruder completely stop.
Is there something that I am missing?

**Reply by Jyers** [reply]
A hardware mod is required to enable linear advance on most boards. Check this for more info on how linear advance can be enabled #707

**Comment by smooreace** [comment]
There is a whole other conversation obout the hoops to jump through to get
LA working. Its not as simple as a firmware change.
